### DISCUSSION TITLE
Is there a way to capture all the msg from the "CypressError" section from the cypress test scripts?

### DISCUSSION THREAD

**Discussion by JingZ-CN** [cypress-io/cypress#17224] [Questions and Help]
Hi guys,

I want to save all the CypressError msg into a test log file during the test execution. And use this log file as test evidence for the future reference. But I am not able to capture those info form the cypress scripts. Do know have any suggestion about this?
![image](https://user-images.githubusercontent.com/30988447/124689521-ff950780-df1b-11eb-92e0-d2d9cecd58ed.png)

**Comment by jennifer-shehane** [comment]
This error message should be available in the `fail` event. Please note that you cannot call `cy.` commands within the `Cypress.on` event listener, so you'll need to use some outside resources to create your log.

```js
Cypress.on('fail', (error, runnable) => {
  // we now have access to the err instance
  // and the mocha runnable this failed on

  throw error // throw error to have test still fail
})
```
